main.py

```diff
@@ -15,41 +12,121 @@
 import re
 import os
 import sys
+import env
+import iitkgp_erp_login.erp as erp
 
 path = os.path.abspath(os.path.dirname(__file__))
 
+# Load professor data
 try:
     with open(os.path.join(path, "data/data.json"), "r") as f:
         profs_dict = CaseInsensitiveDict(json.load(f))
-
 except FileNotFoundError:
     profs_dict = CaseInsensitiveDict({})
 
+# Load department data
+try:
+    with open(os.path.join(path, "data/dept_data"), "r") as f:
+        dept_data = CaseInsensitiveDict(json.load(f))
+except FileNotFoundError:
+    dept_data = CaseInsensitiveDict({})
+
 DEPT_KEY = "dept"
 WEBSITE_KEY = "website"
 TIMETABLE_KEY = "timetable"
 KGP_WEBSITE_URL = "http://www.iitkgp.ac.in/"
-DEPT_FETCH_URL = "http://www.iitkgp.ac.in/facultylist?processOn=onload&colName=&searchContent=&_=1538283022101"
-TIMETABLE_FETCH_URL = (
-    "https://erp.iitkgp.ac.in/Acad/timetable_track.jsp?action=second&dept=%s"
-)
-
+DEPT_FETCH_URL = "https://www.iitkgp.ac.in/Departments/fetchAllFacListByDept"
+TIMETABLE_FETCH_URL = "https://erp.iitkgp.ac.in/Acad/timetable_track.jsp?action=second&dept=%s"
 
 def get_time(slot):
-    # Obtains time for each slot from 'data/slots.1' file
-
+    """Obtains time for each slot from 'data/slots.1' file"""
     with open(os.path.join(path, "data/slots.1")) as f:
         for line in f:
             if line.startswith(slot):
                 return line.split()[1:]
 
+def parse_html(dep, session):
+    """Parses HTML to get professor information and their timetable"""
+    headers = {
+        'Accept': 'application/json, text/javascript, */*; q=0.01',
+        'Accept-Encoding': 'gzip, deflate, br, zstd',
+        'Accept-Language': 'en-GB,en;q=0.9',
+        'Connection': 'keep-alive',
+        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+        'Cookie': 'ci_session=gk7bi3u0r2ej0h3qnihk4gocf0vp9e8e',
+        'Host': 'www.iitkgp.ac.in',
+        'Origin': 'https://www.iitkgp.ac.in',
```

Comment:
Are all of these headers required?

Reply:
Initially I had added some selected headers but it wasn't solving the problem, the data wasn't getting scraped so I copied all the headers, then it started working fine. So, didn't do anything much with that.